### PR TITLE
feat(Lua)!: New Lua bindings.

### LINKED ISSUE #3538: [Bug]: Lua binding for `ImGui.Sliderfloat3` is broken.
### Describe the bug

The binding reads from an empty index.
https://github.com/YimMenu/YimMenu/blob/09b91ca6d719023be7092f7e634e138cdbe36f06/src/lua/bindings/imgui.hpp#L1258
https://github.com/YimMenu/YimMenu/blob/09b91ca6d719023be7092f7e634e138cdbe36f06/src/lua/bindings/imgui.hpp#L1270
https://github.com/YimMenu/YimMenu/blob/09b91ca6d719023be7092f7e634e138cdbe36f06/src/lua/bindings/imgui.hpp#L1282

We need to read from `value[2]` instead of `value[3]`.

### Steps To Reproduce

```lua
local lua_tab = gui.get_tab("GUI_TAB_LUA_SCRIPTS")
local x,y,z

lua_tab:add_imgui(function()
	local input_table = {x,y,z}
	local return_table, used = ImGui.SliderFloat3("Test", input_table, -10, 10)
	if used then
		x = return_table[1]
		y = return_table[2]
		z = return_table[3]
	end
end)
```

### Expected Behavior

`return_table[3]` shouldn't always be empty.

### Logs

[cout.txt](https://github.com/user-attachments/files/16539966/cout.txt)

### Screenshots / Video

_No response_

### Store

Other (please mention in "Additional context")

### Language

English

### YimMenu Language

English (US)

### Additional context

_No response_

### Build

- [X] I've confirmed the issue exists on the latest version of YimMenu

### Coload

- [X] I've tested this without co-loading anything.

### PR DESCRIPTION
**Pull request by tupoy-ya**
###### Terrible title.

- Made an actually independent ImGui binding.
- Added `pointer:set_address` binding.
- Added `menu_event.Wndproc` event.

- Fixed `ImGui.SliderFloat3` binding.
- Fixed some auto generated documentation.

### BREAKING CHANGE
If the path to the script contains "includes" the script will not be loaded.

#### Useless comments
~~- `m_less_dependent_gui` is a terrible name.~~
- This should be multiple pull requests.

Closes #3538

### CODE REVIEW

File: src/lua/bindings/gui.cpp [hunk 1]
```diff
@@ -316,6 +352,33 @@ namespace lua::gui
...
+	//         script.run_in_fiber(function(script)
+	//           -- call natives in there
+	//         end)
+	//       end
+	//
+	//       ImGui.End()
+	//   end
+	// end)
+	// ``
+	static lua::gui::raw_imgui_callback* add_independent_imgui(sol::protected_function imgui_rendering, sol::this_state state)
```
Comment (xiaoxiao921) [inline src/lua/bindings/gui.cpp]: add_always_draw_imgui

File: src/lua/bindings/event.cpp [hunk 2]
```diff
@@ -98,6 +98,18 @@ namespace lua::event
 	// end)
 	// ```
 
+	// Lua API: Field
+	// Table: menu_event
+	// Field: Wndproc: integer
+	// Event that is triggered when we reload the Lua scripts.
```
Comment (xiaoxiao921) [inline src/lua/bindings/event.cpp]: doc is wrong

File: src/lua/lua_manager.cpp [hunk 3]
```diff
@@ -303,6 +316,9 @@ namespace big
 			return {};
 		}
 
+		if(std::filesystem::relative(module_path.parent_path(), m_scripts_folder.get_path()).string().contains("includes"))
```
Comment (xiaoxiao921) [inline src/lua/lua_manager.cpp]: what's the reasoning behind this?

**Review by xiaoxiao921 — CHANGES_REQUESTED** [review]

File: src/lua/lua_manager.cpp (hunk 3 above)
Comment (tupoy-ya) [reply src/lua/lua_manager.cpp]: Some scripts are library scripts, they do nothing on their own and are intended to be used with `require`, they take up space in the script list for no reason.

**Comment by xiaoxiao921** [comment]
```diff
From 149ce7459e30ec6a125fc2351414e3ae0713d186 Mon Sep 17 00:00:00 2001
From: xiaoxiao921 <837334+xiaoxiao921@users.noreply.github.com>
Date: Tue, 13 Aug 2024 15:31:48 +0200
Subject: [PATCH] add reasoning and use native underlying format

---
 src/lua/lua_manager.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/lua/lua_manager.cpp b/src/lua/lua_manager.cpp
index fd54d8b6..756afd93 100644
--- a/src/lua/lua_manager.cpp
+++ b/src/lua/lua_manager.cpp
@@ -316,7 +316,8 @@ namespace big
 			return {};
 		}
 
-		if(std::filesystem::relative(module_path.parent_path(), m_scripts_folder.get_path()).string().contains("includes"))
+		// Some scripts are library scripts, they do nothing on their own and are intended to be used with require, they take up space in the script list for no reason.
+		if (std::filesystem::relative(module_path.parent_path(), m_scripts_folder.get_path()).wstring().contains(L"includes"))
 			return {};
 
 		const auto module_name = module_path.filename().string();
-- 
2.42.0.windows.2


```

**Comment by xiaoxiao921** [comment]
Thanks, I wasnt able to push that commit directly, pr looks good to me! thanks again!

**Review by xiaoxiao921 — APPROVED** [review]

**Pull request merged**